wave_optimization/include/wave/optimization/ceres/graph_wrapper.hpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file
+ * @ingroup optimization
+ * Ceres wrappers for FactorGraph classes
+ *
+ * @todo reorganize - move to another file
+ * @todo build up ceres Problem simultaneously while building up FactorGraph
+ */
+#ifndef WAVE_OPTIMIZATION_CERES_GRAPH_WRAPPER_HPP
+#define WAVE_OPTIMIZATION_CERES_GRAPH_WRAPPER_HPP
+
+#include <ceres/ceres.h>
+#include <functional>
+#include "wave/optimization/factor_graph/FactorGraph.hpp"
+
+namespace wave {
+/** @addtogroup optimization
+ *  @{ */
+
+class FactorBase;
+class FactorGraph;
+
+
+class FactorGraphProblem : public ceres::Problem {
```

Comment:
Could we avoid this inheritance? I don't think I see this `FactorGraphProblem` class being used polymorphically anywhere, so it might be better to just make `ceres::Problem` a member variable.